opta/core/helm_cloud_client.py

```diff
@@ -80,38 +81,60 @@ def get_all_remote_configs(self) -> Dict[str, Dict[str, "StructuredConfig"]]:
 
     def set_kube_config(self) -> None:
         kube_config_file_name = self.layer.get_kube_config_file_name()
-        default_kube_config_filename = expanduser(
-            constants.DEFAULT_KUBECONFIG.split(ENV_KUBECONFIG_PATH_SEPARATOR)[0]
-        )
+        default_kube_config_filename = constants.DEFAULT_KUBECONFIG.split(
+            ENV_KUBECONFIG_PATH_SEPARATOR
+        )[0]
         if not exists(default_kube_config_filename):
             raise UserErrors(
                 "To use K8s-native provider, your must have your kubeconfig set"
             )
         logger.debug("Loading kube config file")
         try:
-            default_kube_config = yaml.load(open(default_kube_config_filename))
-        except Exception:
+            with open(default_kube_config_filename) as f:
+                default_kube_config = yaml.load(f)
+        except YAMLError:
```

Comment:
```suggestion
        except yaml.YAMLError:
```

(as opposed to `from ruamel.yaml import YAMLError` above). The `yaml` util library should be used instead of importing ruamel yaml directly, to discourage unsafe usage. That library reexports YAMLError.